src/som/compiler/Variable.java

```diff
@@ -57,6 +57,9 @@ public boolean equals(final Object o) {
       assert name.equals(var.name) : "Defined in the same place, but names not equal?";
       return true;
     }
+    if (var.name == "self" && name == "self") {
```

Comment:
So, you are saying that the source information for `self` nodes aren't correct?
I'd try to fix those, instead of adding the hack here (which is also not correct/reliable Java as the codacy bot indicates).
It could well be that there are issues with `self` since there are nodes constructed somewhat manually in various places, which is something not happening with other variables.